ros2controlcli/ros2controlcli/verb/set_hardware_component_state.py

```diff
@@ -0,0 +1,127 @@
+                if matched_hardware_components.state.label != "inactive":
+                    return (
+                        f"cannot cleanup {matched_hardware_components.name} "
+                        f"from its current state {matched_hardware_components.state}"
+                    )
+
+                unconfigured_state = State()
+                unconfigured_state.id = State.PRIMARY_STATE_UNCONFIGURED
+                unconfigured_state.label = "unconfigured"
+
+                response = set_hardware_component_state(
+                    node, args.controller_manager, args.hardware_component_name, unconfigured_state
+                )
+                if not response.ok:
+                    return "Error cleaning up hardware component, check controller_manager logs"
+
+                print(f"Successfully set {args.hardware_component_name} to state {response.state}")
+                return 0
+
+            if args.state == "inactive":
+                inactive_state = State()
+                inactive_state.id = State.PRIMARY_STATE_INACTIVE
+                inactive_state.label = "inactive"
+
+                if matched_hardware_components.state.label == "unconfigured":
+                    response = set_hardware_component_state(
+                        node, args.controller_manager, args.hardware_component_name, inactive_state
+                    )
+                    if not response.ok:
+                        return (
+                            "Error configuring hardware component, check controller_manager logs"
+                        )
+
+                    print(
+                        f"Successfully set {args.hardware_component_name} to state {response.state}"
+                    )
+                    return 0
+
+                elif matched_hardware_components.state.label == "active":
+                    response = set_hardware_component_state(
+                        node, args.controller_manager, args.hardware_component_name, inactive_state
+                    )
+                    if not response.ok:
+                        return "Error stopping hardware component, check controller_manager logs"
+
+                    print(
+                        f"Successfully set {args.hardware_component_name} to state {response.state}"
+                    )
+                    return 0
+
+                else:
+                    return (
+                        f'cannot put {matched_hardware_components.name} in "inactive" state '
+                        f"from its current state {matched_hardware_components.state}"
+                    )
+
+            if args.state == "active":
+                if matched_hardware_components.state.label != "inactive":
+                    return (
+                        f"cannot activate {matched_hardware_components.name} "
+                        f"from its current state {matched_hardware_components.state}"
+                    )
```

Comment:
This logic is also written in the ResourceManager, should we just let any request come through and then return here error or success with a message. Not sure if doubling logic is good for maintainability, but it is good for efficiency. Still, not sure that this degree of efficiency is needed here.

Reply:
I also think that we shouldn't duplicate the logic,
But I want to avoid is something like with other controller-manager services, where one log-line is an error from controller-manager but the service returns OK. I'll have a look if `response.ok` is fine for giving useful feedback with the CLI

Reply:
I removed the logic now..